packages/devtools_app/lib/src/profiler/flutter/cpu_profile_flame_chart.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/material.dart';
+
+import '../../charts/flutter/flame_chart.dart';
+import '../../ui/colors.dart';
+import '../../utils.dart';
+import '../cpu_profile_model.dart';
+
+class CpuProfileFlameChart extends FlameChart<CpuProfileData, CpuStackFrame> {
+  CpuProfileFlameChart(
+    CpuProfileData data, {
+    @required double width,
+    @required CpuStackFrame selected,
+    @required Function(CpuStackFrame stackFrame) onSelected,
+  }) : super(
+          data,
+          duration: data.profileMetaData.time.duration,
+          totalStartingWidth: width,
+          startInset: sideInsetSmall,
+          endInset: sideInsetSmall,
+          selected: selected,
+          onSelected: onSelected,
+        );
+
+  @override
+  _CpuProfileFlameChartState createState() => _CpuProfileFlameChartState();
+}
+
+class _CpuProfileFlameChartState
+    extends FlameChartState<CpuProfileFlameChart, CpuStackFrame> {
+  static const stackFramePadding = 1;
+
+  int _colorOffset = 0;
+
+  @override
+  void initFlameChartElements() {
+    super.initFlameChartElements();
+    expandRows(widget.data.cpuProfileRoot.depth +
+        rowOffsetForTopPadding +
+        rowOffsetForBottomPadding);
+
+    final Map<String, double> stackFrameLefts = {};
+
+    double leftForStackFrame(CpuStackFrame stackFrame) {
+      final CpuStackFrame parent = stackFrame.parent;
+      double left;
+      if (parent == null) {
+        left = widget.startInset;
+      } else {
+        final stackFrameIndex = stackFrame.index;
+        if (stackFrameIndex == 0) {
+          // This is the first child of parent. [left] should equal the left
+          // value of [stackFrame]'s parent.
+          left = stackFrameLefts[parent.id];
+        } else {
+          assert(stackFrameIndex != -1);
+          // [stackFrame] is not the first child of its parent. [left] should
+          // equal the right value of its previous sibling.
+          final CpuStackFrame previous = parent.children[stackFrameIndex - 1];
+          left = stackFrameLefts[previous.id] +
+              (widget.startingContentWidth * previous.totalTimeRatio);
+        }
+      }
+      stackFrameLefts[stackFrame.id] = left;
+      return left;
+    }
+
+    void createChartNodes(CpuStackFrame stackFrame, int row) {
+      final double width =
+          widget.startingContentWidth * stackFrame.totalTimeRatio -
+              stackFramePadding;
+      final left = leftForStackFrame(stackFrame);
+      final backgroundColor = _colorForStackFrame(stackFrame);
+
+      final node = FlameChartNode<CpuStackFrame>(
+        key: Key('${stackFrame.id}'),
+        text: stackFrame.name,
+        tooltip: '${stackFrame.name} - ${msText(stackFrame.totalTime)}',
+        rect: Rect.fromLTRB(left, flameChartNodeTop, left + width, rowHeight),
```

Comment:
comment why the bottom is rowHeight even though top is nonZero. That looks more intuitively like fromLRWH but I am probably missing something.

Reply:
modified to use LTWH as that is more intuitive.